extensions/cl_khr_unified_svm.asciidoc

```diff
@@ -0,0 +1,1081 @@
++
+--
+`RESOLVED`:
+The initial version of this extension will not support larger fill patterns.
+--
+
+. Can a pointer to a device, host, or shared SVM allocation be used to create a `cl_mem` using `CL_MEM_USE_HOST_PTR`?
++
+--
+*UNRESOLVED*:
+Trending "no" in all cases.
+If the SVM allocation is from the same context this could be an error, such as `CL_INVALID_HOST_PTR`.
+If the SVM allocation is from a different context then behavior could be undefined.
+--
+
+. Can a pointer to a device, host, or shared SVM allocation be used to create a `cl_mem` buffer using `CL_MEM_COPY_HOST_PTR`?
++
+--
+*UNRESOLVED*:
+Trending "no" for device and shared USM allocations.
+If the USM allocation is from the same context this could be an error, such as `CL_INVALID_HOST_PTR`.
+If the USM allocation is from a different context then behavior could be undefined.
+
+Trending "yes" for host USM allocations, both when the host USM allocation is from this context and from another context.
+--
+
+. Can a pointer to a device, host, or shared SVM allocation be passed to API functions to read from or write to `cl_mem` objects, such as *clEnqueueReadBuffer* or *clEnqueueWriteImage*?
++
+--
+*UNRESOLVED*:
+Trending "yes" for device SVM allocations, so long as the device SVM allocation is accessible by the device associated with the command-queue, and the device allocation was made against the context associated with the command-queue.
+
+Trending "yes" for host USM allocations, both when the host USM allocation is from this context and from another context.
+
+Trending "no" for shared USM allocations.
+If the shared USM allocation is from the same context this could be an error, such as `CL_INVALID_HOST_PTR`.
+If the shared USM allocation is from a different context then behavior could be undefined.
+--
+
+. Can a pointer to a device, host, or shared USM allocation be passed as the `pattern` argument to API functions to fill a `cl_mem`, SVM allocation, or USM allocation, such as *clEnqueueFillBuffer*?
++
+--
+*UNRESOLVED*:
+Trending "no" for device and shared allocations.
+If the USM allocation is from the same context this could be an error, such as `CL_INVALID_HOST_PTR`.
+If the USM allocation is from a different context then behavior could be undefined.
+
+Trending "yes" for host USM allocations, both when the host USM allocation is from this context and from another context.
+--
+
+. Should we support passing traditional `cl_mem_flags` via the USM allocation properties?
++
+--
+`RESOLVED`: We decided not to accept any `cl_mem_flags` or `cl_svm_mem_flags`, and added access properties instead.
+--
+
+. Exactly how do the additional SVM types affect the memory model?
++
+--
+*UNRESOLVED*:
+--
+
+. Should it be an error to set an unknown pointer as a kernel argument using *clSetKernelArgSVMPointer* if no devices support shared system allocations?
```

Comment:
Discussed in the January 7th memory subgroup.  For SYCL, we generally want to follow C++ semantics, meaning that it is valid to pass any pointer to a function (or kernel) and it is only an problem (specifically, undefined behavior) to dereference a bogus pointer.  This would imply that it should NOT be an error to set a kernel argument that does not point into a memory region returned by clSVMAlloc, even for an implementation that didn't support system SVM.

See: https://github.com/KhronosGroup/OpenCL-Docs/pull/905

Reply:
Discussed in the July 1st memory subgroup.  Given that we recently updated the Intel USM extension to follow these C++ semantics at the request of our SYCL runtime, I strongly recommend that we adopt this behavior for the unified SVM extension also.  This means it would not be an error to set a bogus kernel argument and only would be an error to dereference a bogus pointer in the kernel.

Reply:
This is resolved in the spec document and is tested in the CTS.  It is not an error to set an unknown pointer as a kernel argument and it is only an error to dereference the unknown pointer in a kernel.  Therefore, I am resolving this GitHub comment also.